### PR TITLE
feat: adds `watchFragment` support

### PR DESCRIPTION
**Pull request by alessbell**
Adds support for a new method available on an `ApolloClient` in 3.10, `watchFragment` ([docs](https://deploy-preview-11698--apollo-client-docs.netlify.app/api/core/ApolloClient#watchfragment)).

## To do
- [ ] docs
- [x] update the Apollo Client version once 3.10 is released

### CODE REVIEW

File: packages/apollo-angular/src/fragment.ts
Comment (PowerKiKi) [inline packages/apollo-angular/src/fragment.ts:13]:
```suggestion
  public watchFragment(options?: WatchFragmentOptionsAlone<T, V>): Observable<WatchFragmentResult<T>> {
```

This should use a new type `WatchFragmentOptionsAlone` that exclude `fragment` and `from`. And those tow required options should instead be delcared as inheritable members, as is done for `document` in the equivalent `Query` class.

But all of this really makes sense if are able to generate code for fragments too. Is is something you planned ?

Comment (alessbell) [reply packages/apollo-angular/src/fragment.ts]: Thanks @PowerKiKi - I've made part of that change in https://github.com/kamilkisiela/apollo-angular/pull/2226/commits/5978d22deeb7baa6d3547a0f1d5ae736eb3eda79, though after thinking about it a bit more I think it makes sense to exclude `fragment` but not `from`. I imagine that a Fragment service would be created once per fragment document, but that you may want to pass in the object reference via `from` at the call site. An example might be helpful here - would it be possible to create a one-off release from this PR?

Also re: the codegen story for this feature, I was discussing this with @phryneas a bit and he had some ideas around an e.g. `Fragment.from(document)` first-class API instead. Should I create a separate issue for that discussion?

**Comment by chimurai** [comment]
Looking forward to use `watchFragment` 🙂

Is help needed to land this PR?

([apollo-client 3.10](https://github.com/apollographql/apollo-client/releases/tag/v3.10.0) has been released)

File: packages/apollo-angular/src/fragment.ts
Comment (PowerKiKi) [reply packages/apollo-angular/src/fragment.ts]: It seems I forgot to reply earlier. Sorry about that.

Keeping `from` as an option makes a lot of sense 👍

I'm not sure what the hypothetical `Fragment.from(document)` means. Feel free to explain more here, or in a new thread if you feel like the discussion might become too big/unrelated.

**Comment by alessbell** [comment]
Hi @chimurai 👋 Glad to know you're excited for this feature :) I'll get the PR updated and ready for final review.

**Comment by alessbell** [comment]
Not sure why that one test timed out - @PowerKiKi any ideas there?

**Comment by alessbell** [comment]
It looks like these tests are failing on `main` - anything I can do to help unblock this PR, @PowerKiKi?

**Comment by PowerKiKi** [comment]
It's fine if the docs comes in the next PR.

I fixed the CI that broke because of Renovate in the unrelated #2267.

Before merging this, I'd like to understand the story around the newly introduced class `Fragment`. I think it only makes sense if it is used by code generations tools. What's your opinion ? Will `@graphql-codegen/cli` be adapted accordingly ? or should we drop the class ? or is it useful for something else ?

**Comment by alessbell** [comment]
Thanks @PowerKiKi - I'll descope codegen for now and have gone ahead and removed the Fragment class from the PR. Let me know if there's anything else, appreciate the reviews!

**Review by PowerKiKi — APPROVED** [review]
LGTM. If you can squash, I'll merge.

**Comment by alessbell** [comment]
Squashed and needed to merge master one last time - lmk if that's an issue. Thanks again @PowerKiKi.

**Comment by PowerKiKi** [comment]
Thanks for your work. It's been released as 7.1.1 (7.1.0 release process crashed)

**Pull request merged**
